package/MDAnalysis/topology/TXYZParser.py

```diff
@@ -88,9 +91,22 @@ def parse(self, **kwargs):
             names = np.zeros(natoms, dtype=object)
             types = np.zeros(natoms, dtype=np.int)
             bonds = []
+            # Find first atom line, maybe there's box information
+            fline = inf.readline()
+            try:
+                # If a box second value will be a float
+                # If an atom, second value will be a string
+                float(fline.split()[1])
+            except ValueError:
+                # If float conversion failed, we have first atom line
+                pass
+            else:
+                # If previous try succeeded it was a box
+                # so read another line to find the first atom line
+                fline = inf.readline()
             # Can't infinitely read as XYZ files can be multiframe
-            for i in range(natoms):
-                line = inf.readline().split()
+            for i, line in zip(range(natoms), itertools.chain([fline], inf)):
```

Comment:
The linter was complaining about this line because it thought we weren't iterating the `range`, but we are, so I changed the linter RC file.